src/Php/PolyfillPackagesProvider.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Php;
+
+use Nette\Utils\FileSystem;
+use Nette\Utils\Json;
+use Rector\Core\Configuration\Option;
+use Rector\Core\Configuration\Parameter\SimpleParameterProvider;
+use Rector\Core\ValueObject\PolyfillPackage;
+
+final class PolyfillPackagesProvider
+{
+    /**
+     * @var array<PolyfillPackage::*>
+     */
+    private array $cachedPolyfillPackages = [];
```

Comment:
This should be nullable,
- cache is null: never filled
- cache is array, it filled, even empty, it means already read composer.json

Reply:
Makes sense, could you add it?

Reply:
sure, I created PR :

- https://github.com/rectorphp/rector-src/pull/5390

for it.